### PR TITLE
[coordinator] Support additional metric types on remote write

### PR DESCRIPTION
**Pull request by shreyassrivatsan**
**What this PR does / why we need it**:
Adds support for gauges and timers in the remote write path. Allowing us to pass them to aggregator with the appropriate aggregator metric.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
```release-note
Adds support for gauge and timer metric types in coordinator so that we can pass through the different aggregator supported types.
```

**Does this PR require updating code package or user-facing documentation?**:
```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/services/m3coordinator/ingest/write_test.go [hunk 1]
```diff
@@ -366,6 +382,51 @@ func TestDownsampleAndWriteBatch(t *testing.T) {
...
+			session.EXPECT().WriteTagged(
+				gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), dp.Value, gomock.Any(), entry.annotation,
+			)
+		}
+	}
+
+	iter := newTestIter(testEntries2)
+	err := downAndWrite.WriteBatch(context.Background(), iter, WriteOptions{})
+	require.NoError(t, err)
+}
```
Comment (robskillington) [inline src/cmd/services/m3coordinator/ingest/write_test.go:429]: nit: Extra endline?

File: src/query/api/v1/handler/prometheus/remote/write.go [hunk 2]
```diff
@@ -483,7 +483,12 @@ func (h *PromWriteHandler) write(
 	r *prompb.WriteRequest,
 	opts ingest.WriteOptions,
 ) ingest.BatchError {
-	iter := newPromTSIter(r.Timeseries, h.tagOptions)
+	iter, err := newPromTSIter(r.Timeseries, h.tagOptions)
+	if err != nil {
+		var errs xerrors.MultiError
+		errs.Add(err)
```
Comment (robskillington) [inline src/query/api/v1/handler/prometheus/remote/write.go:489]: This should be `errs = errs.Add(err)`?

File: src/query/ts/metadata.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+const (
+	// SourceTypePrometheus is the prometheus source type.
+	SourceTypePrometheus SourceType = iota
+
+	// SourceTypeGraphite is the graphite source type.
+	SourceTypeGraphite
+)
+
+// SeriesInfo is information about the time series.
+type SeriesInfo struct {
```
Comment (robskillington) [inline src/query/ts/metadata.go:49]: nit: SeriesAttributes or SeriesMetdata could also be alternative names, we already have `storage.Attributes` for each write (retention/resolution). What do you think? Not super bullish, but info always seems so generic.

Comment (shreyassrivatsan) [reply src/query/ts/metadata.go:49]: `SeriesAttributes` is good. Will change to that.

File: src/cmd/services/m3coordinator/ingest/write.go [hunk 4]
```diff
@@ -45,7 +45,7 @@ var (
 // the WriteBatch method.
 type DownsampleAndWriteIter interface {
 	Next() bool
-	Current() (models.Tags, ts.Datapoints, xtime.Unit, []byte)
+	Current() (models.Tags, ts.Datapoints, ts.SeriesAttributes, xtime.Unit, []byte)
```
Comment (arnikola) [inline src/cmd/services/m3coordinator/ingest/write.go:48]: nit: pull into a type?

**Pull request merged**